src/pages/admin/student/edit/page.tsx

```diff
@@ -1,17 +1,58 @@
 import * as React from 'react';
-import {Link} from 'react-router';
-import {adminRouteEnums} from '../../../../common/routeEnums/admin';
-
-export class EditStudentPage extends React.Component<{}, {}> {
-   public render() {
-       return (
-         <div>
-           <span> Edit Student Page: </span>
-           <br/>
-           <br/>
-           <Link to={adminRouteEnums.student.list}>Back to student list</Link>
-           <Link to={adminRouteEnums.default}>Back to Dashboard</Link>
-         </div>
-        );
+import { Link } from 'react-router';
+import { Student } from '../../../../model/student';
+import { EditStudentComponent } from './components/editStudentComponent';
+
+interface IProps extends React.Props<EditStudentPage> {
+  student: Student;
+  getStudent: (id: number) => void;
+  saveStudent: (student: Student) => void;
+}
+
+export class EditStudentPage extends React.Component<IProps, {}> {
+  public componentDidMount() {
+    // TODO: Get the Student Id from the URL
+    this.props.getStudent(32);
   }
+
+  public render() {
+    if (this.props.student.id === -1) {
```

Comment:
I guess you will add here something like insert student? 

About the "-1" we will have to enhance this, playing with this number is a bit dangerous.

Insert mode is going to be supported? we can write here something like "insert mode coming soon", if not, if it's an error message we can extract a subcomponent.